projenrc/github-workflows.ts

```diff
@@ -493,3 +493,163 @@ export function runCommitLintWorkflow(project: AwsCdkConstructLibrary) {
     }
   }
 }
+
+/**
+ * Runs the code generation step to update the list of available JumpStart models, and DLC images.
+ * Uses https://github.com/aws-actions/configure-aws-credentials to authenticate against AWS
+ * @param project AwsCdkConstructLibrary
+ */
+export function buildCodeGenerationWorkflow(project: AwsCdkConstructLibrary) {
+
+  const CREATE_PATCH_STEP_ID = 'create_patch';
+  const PATCH_CREATED_OUTPUT = 'patch_created';
+
+  // First job to run the code generation step
+  const generate: Job = {
+    name: 'Code generation',
+    runsOn: ['ubuntu-latest'],
+    permissions: {
+      idToken: JobPermission.WRITE, //needed to interact with GitHub's OIDC Token endpoint.
+      contents: JobPermission.READ,
+    },
+    outputs: {
+      [PATCH_CREATED_OUTPUT]: {
+        stepId: CREATE_PATCH_STEP_ID,
+        outputName: PATCH_CREATED_OUTPUT,
+      },
+    },
+    steps: [
+      {
+        name: 'Checkout project',
+        uses: 'actions/checkout@v3',
+        with: {
+          ref: 'main',
+        },
+      },
+      {
+        name: 'Setup Node.js',
+        uses: 'actions/setup-node@v3',
+        with: {
+          'node-version': '20.x',
+        },
+      },
+      {
+        name: 'Install dependencies',
+        run: 'yarn install --check-files --frozen-lockfile',
+      },
+      {
+        name: 'Setup AWS credentials',
+        uses: 'aws-actions/configure-aws-credentials@v3.0.1',
+        with: {
+          'role-to-assume': '${{ secrets.AWS_ROLE_ARN_TO_ASSUME }}',
+          'aws-region': '${{ env.AWS_REGION }}',
+          'role-duration-seconds': '7200',
+        },
+      },
+      {
+        name: 'Run code generation',
+        run: 'npx projen generate-models-containers',
+      },
+      {
+        name: 'Find mutations',
+        id: CREATE_PATCH_STEP_ID,
+        run: [
+          'git add .',
+          'git diff --staged --patch --exit-code > .repo.patch || echo "patch_created=true" >> $GITHUB_OUTPUT',
+        ].join('\n'),
+      },
+      {
+        name: 'Upload patch',
+        if: `steps.${CREATE_PATCH_STEP_ID}.outputs.${PATCH_CREATED_OUTPUT}`,
+        uses: 'actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32',
```

Comment:
Use `v3` as the `.projenrc.ts` replaces these with specific commit hashes: https://github.com/awslabs/generative-ai-cdk-constructs/blob/561222d466d3f3b72d16bee6dd425f7ea659aaa6/.projenrc.ts#L127
```suggestion
        uses: 'actions/upload-artifact@v3',
```

Reply:
done in [082cc46](https://github.com/awslabs/generative-ai-cdk-constructs/pull/238/commits/082cc46cc0456acc6557d153b746b9921785ed63)